### ISSUE TITLE
Feature request: Add to cart

### ISSUE THREAD

**Issue by moroz** [JocelynH1110/self-order#4]
In the program loop, implement a new command `add`. This command accepts two arguments, one required and one optional:
```
> add PRODUCT_ID [QUANTITY]
```
If `QUANTITY` is not provided, it should default to 1.

If the product is not in cart, add a new item to the cart. If the same product has already been added to the cart, increment the quantity of the existing cart item.

Update the `cart` command to display the contents of the cart, if any:

```
> cart
Your cart is currently empty.
> add 1 2
> cart
You have 1 item(s) in the shopping cart.
1. Latte, 2 x $60 = $120
```

Implementation suggestion: Implement a new **data type** called `CartItem`. You can store the user's cart in a variable: either local to the `main()` function, or a package-level global variable. (e. g. `var items = make([]CartItem, 0)`). Or you can use a hash map, mapping product ids to quantities.
You may also implement a `Cart` struct embedding multiple `CartItem`s and add or remove these items using a method implemented on the `Cart` struct.
The `CartItem` struct should contain a product's ID and quantity. This is because at a later stage we are going to store these cart items in a relational database.

Ultimately, the choice of the underlying data structures is up to you!

**Comment by moroz** [comment]
## Adding item to cart, implementing methods on the Cart struct

In the method that adds a product to cart (the method on the `Cart` type), you need to first check if the product has already been added to the cart. You can do this in another method, for example one with a signature like this:
```go
func (c *Cart) findProductInCart(productID int) int
```
This function would return the index of the cart item if there is a cart item with this `productID`, or `-1` if there is none.
In order to find an item in a slice, you need to iterate over all items in the slice:
```go
for i, item := range c.CartItems {
  // if the productID matches, return index
}
// handle no item found
```

Then, when adding an item to cart, you can check it like this:
```go
index := c.findProductInCart(productID)
if index >= 0 {
  // update the Quantity on the existing item
} else {
  // create a new cart item and append it to existing CartItems
}
```

When updating an existing item, you need to use the indexing operator (e. g. `c.CartItems[index]`), and when you add a new item, you need to use the [`append`](https://golangbyexample.com/append-function-in-golang/) built-in function.
For now, the cart can be a package-level global variable, e. g.:

```go
package main

var cart = Cart{} // I'm not sure if this will work
```

## Parsing commands

In `handleCommand`, you split the command into _tokens_, with each token separated by spaces.
If the command is `add`, you can validate (驗證) the arguments to the `add` command.

E. g. if the command is passed like this:

```
> add 1 2
```
Then the `tokens` will be a `[]string` with 3 elements:
```
[]string{"add", "1", "2"}
```
In order to validate the command, you need to check that:

* there is one or two arguments (`len(tokens) == 2 || len(tokens) == 3`).
* the first and second arguments can be parsed as `int` and are both positive (`> 0`).
* the first argument is the `productID`, check that there exists a product with this `ID`.

**Comment by moroz** [comment]
Resolved by #10
